### ISSUE TITLE
Milkmaid stopped working 1.19.4

### ISSUE THREAD

**Issue by Pretecha** [Alex-the-666/Rats#801]
An issue where the milkmaid stops completely. (Apparently it has happened before, but I don't think anyone has brought up this issue in this version)

Behaviour: The rat is given the milkmaid upgrade. As soon as the command is applied to harvest, the rat races to the cow and attempt to milk it. It then just stops there and doesn't do anything else unless changed command or an item is dropped. Same thing happens when you apply command first and give the upgrade. The deposit is set before both actions.

I've tried probably every way to fix it. Changing the order of applying(milkmaid first, command second, deposit route third, etc), giving it an empty/milk bucket, changing placement of cow/curdling station, and even whitelisting milk bucket and empty buckets...

**Comment by HeatSignature** [comment]
I would say there is quite a lot of the AI behavior in this mod that needs reworking in 19.4. Breeders won't even breed chickens because they pick up eggs on the ground which prevents them holding seed/wheat etc. Replanting/planting also doesn't work properly(in my pack). Hopefully they are still working on it.

**Comment by GizmoTheMoonPig** [comment]
I am, don't worry! I've been struggling to find time to work on the mod but I'm doing my best. I'm gonna have to do another pass on work goals, it seems like quite a few of them are still a bit buggy. This mod was in a really bad state when I picked it up so please have patience with me. The mod is still in its beta stage for a reason!

Do note that because I'm already limited on time all future fixes will be done in the upcoming 1.20 port, since I stick with working on the latest MC version for all my mods. Hope that doesn't bum anyone out too much, I picked a really bad time to start porting this mod and I'm hoping 1.20 will be a more popular version to use.

**Comment by GizmoTheMoonPig** [comment]
ive had a milkmaid rat running for about an hour now and its been working just fine. This may be another case of me fixing something and it not making it into another beta. I do recall expanding the interaction radius for animals at some point (includes both breeding and milking) so that may have fixed it. I did also make it so the rat won't climb on top the auto curdler which seemed to halt pathfinding for some reason. I will be releasing a beta on 1.20 in the upcoming days, hope you're willing to try it out and see if things work better for you! 

> Breeders won't even breed chickens because they pick up eggs on the ground which prevents them holding seed/wheat etc.

I'll be pushing out a fix to this now. This fix will allow upgrades to decide whether rats should pick up certain items off the floor. I disabled all item pickups for the breeder, it will still grab items from chests but not off the ground.
